application_sandbox/sample_application_framework/sample_application.h

```diff
@@ -26,7 +26,8 @@ namespace sample_application {
 
 const static VkSampleCountFlagBits kVkMultiSampledSampleCount =
     VK_SAMPLE_COUNT_4_BIT;
-const static VkFormat kDepthFormat = VK_FORMAT_D16_UNORM;
+const static VkFormat kDepthFormatFixed = VK_FORMAT_D16_UNORM;
+const static VkFormat kDepthFormatFloat = VK_FORMAT_D32_SFLOAT;
```

Comment:
VK_FORMAT_D32_SFLOAT Is not guaranteed to be supported on all hardware (D16 is), so we will end up just failing on creation with weird errors.

If we want to allow EXPLICITLY floating point (rather than a higher precision, D24_UNORM) we need to make device selection depend on this.